app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class CartsController < ApplicationController
+  before_action :initialize_session
+  decorates_assigned :cart_items
+
+  def index
+    @cart_items = Menu.where(id: session[:cart])
+  end
+
+  def create
+    AddToCartService.call(menu_id: params['cart']['id'].to_i,
+                          session_cart: session[:cart])
+    redirect_to users_dashboard_index_path
+  end
+
+  def destroy
+    session[:cart].delete(params['id'].to_i)
```

Comment:
cannot see, why you are using string keys to access params, instead of `params[:id]` :)
also, I'm not sure this `.to_i` all over the place is necessary. You could work with strings in the first place